salt/metalk8s/volumes/prepared/init.sls

```diff
@@ -0,0 +1,51 @@
+include:
+  - .installed
+
+Refresh pillars for all minions:
+  salt.function:
+    - name: saltutil.refresh_pillar
+    - tgt: '*'
+
+{%- set volumes_to_create = [] %}
```

Comment:
Instead of this, could we have a `single` SLS which requires a single volume name to be present and process it (which can then be called from the operator through SaltAPI), and have this one loop over everything, and include `single` for every loop iteration with the loop value in the context as `volume`?

I.e. instead of supporting both handling one single volume and all of them in the same SLS/Jinja template, split it up to make things easier to comprehend.

Reply:
But you can't pass information between state.
You only can with orchestrate with pillar but I think we don't want to have an orchestrate to prepare volume as this definitly not need any orchestration

Reply:
Is there no way to 'include' another SLS with a given context dict to render that sub-SLS before pasting in the result?

Reply:
Not as far as I know, I think it's good question for saltstack slack :)

Reply:
Seems not supported but asked since long time https://github.com/saltstack/salt/issues/8878

Reply:
I guess we will have to keep it like this then, right?

Reply:
Yes I think so, except if someone else have an idea

Reply:
No, except writing code. That's really too bad.

Reply:
So I'll resolve the conversation.

But yes, it's a pity that we can't "reuse" easily a templated SLS into another one :(

Reply:
(Or using a Jinja macro, but bleh)